From 4c72659b726703f9d8b60429456831ff0659b6fe Mon Sep 17 00:00:00 2001
From: Alex Williamson <alex.williamson@redhat.com>
Date: Fri, 6 Jun 2014 16:28:01 +0200
Subject: [PATCH 2/2] Set PSH flag only on packets containing data

RH-Author: Alex Williamson <alex.williamson@redhat.com>
Message-id: <20140606162801.23875.14538.stgit@bling.home>
Patchwork-id: 59171
O-Subject: [RHEL6.6 gpxe PATCH 2/2] [tcp] Set PSH flag only on packets containing data
Bugzilla: 1105189
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

From: Michael Brown <mcb30@ipxe.org>

Bugzilla: 1105189
Upstream commit: 67dc832d152bbbaa6c3046eb68e876d2527beb9a (ipxe.git)

Suggested-by: Yelena Kadach <klenusik@hotmail.com>
Signed-off-by: Michael Brown <mcb30@ipxe.org>
---
 src/net/tcp.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Signed-off-by: jen <jen@redhat.com>
---
 src/net/tcp.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/net/tcp.c b/src/net/tcp.c
index 0ef6577..574900b 100644
--- a/src/net/tcp.c
+++ b/src/net/tcp.c
@@ -471,7 +471,7 @@ static int tcp_xmit ( struct tcp_connection *tcp, int force_send ) {
 		tsopt->tsopt.tsval = ntohl ( currticks() );
 		tsopt->tsopt.tsecr = ntohl ( tcp->ts_recent );
 	}
-	if ( ! ( flags & TCP_SYN ) )
+	if ( len != 0 )
 		flags |= TCP_PSH;
 	tcphdr = iob_push ( iobuf, sizeof ( *tcphdr ) );
 	memset ( tcphdr, 0, sizeof ( *tcphdr ) );
-- 
1.7.1

